rasterio/plot.py

```diff
@@ -33,38 +33,69 @@
 logger = logging.getLogger(__name__)
 
 
-def show(source, cmap='gray', with_bounds=True):
+def show(source, cmap='gray', with_bounds=True, ax=None, title=None, **kwargs):
     """Display a raster or raster band using matplotlib.
 
     Parameters
     ----------
-    source : array-like or (raster dataset, bidx)
+    source : array-like, or (raster dataset, bidx) tuple, or raster dataset, 
         If array-like, should be of format compatible with
         matplotlib.pyplot.imshow. If the tuple (raster dataset, bidx),
```

Comment:
Absolutely, I think this implementation detail should be hidden from users as much as possible.  I think it would be nice if it seamlessly handled either raster orientation if possible. Is there a convenient way to check if a array is in raster shape or image shape?  

It doesn't seem ironclad but maybe something like:

def raster_orientation(arr):
    if arr.shape[0] > arr.shape[1]:
        return "image"
    else:
        return "raster"